### PR TITLE
fix(glv): don't assume d1 or d2 are in the field + GLV fixes (morally)

### PR DESCRIPTION
**Pull request by kobigurk**

### CODE REVIEW

File: algebra-core/src/fields/macros.rs [hunk 1]
```diff
@@ -210,6 +210,17 @@ macro_rules! impl_Fp {
                 }
             }
 
+            #[inline]
+            fn from_repr_unchecked(r: $BigIntegerType) -> Option<Self> {
```
Comment (jon-chuang) [inline algebra-core/src/fields/macros.rs]: This doesn't need to be option, does it?

Also, why define a function that isn't being used anywhere?

Comment (jon-chuang) [reply algebra-core/src/fields/macros.rs]: Unless you meant to use it in glv.rs

**Review by jon-chuang — COMMENTED** [review]
The restriction that the input is smaller than modulus is theoretically unnecessary. Rather, one needs it to be smaller than R, which for our purposes is a power of 2, since we need T, the product, to be in the range [0, R*N]. However, the validity of the input would then be dependent on the choice of R. Realistically, one could have always chosen `R = 2^64 * n_limbs`, but I don't think this is the case. Unfortunately, what in my opinion is a poorer choice, the next power of 2 larger than the modulus, is chosen.

This is the impl. So yeah, it's wrong, I think.
```
  /// Let `M` be the power of 2^64 nearest to `Self::MODULUS_BITS`. Then
  /// `R = M % Self::MODULUS`.
  const R: Self::BigInt;
```

Nevertheless, for our particular application, we have the invariant that `d_i < 2 * n`, so realistically, one should just test if its greater than modulus and if so subtract, and use old method.

File: algebra-core/src/fields/macros.rs (hunk 1 above)
Comment (kobigurk) [reply algebra-core/src/fields/macros.rs]: I committed the wrong version. This version is slow.

Comment (kobigurk) [reply algebra-core/src/fields/macros.rs]: @jon-chuang take a look now?

**Comment by jon-chuang** [comment]
actually this is also eroneous, morally speaking. Let me PR the right fix

**Review by jon-chuang — APPROVED** [review]
Changes are straightforward and address the discovered issues

**Pull request merged**